itest/lnd_mpp_test.go

```diff
@@ -13,10 +14,124 @@ import (
 	"github.com/lightningnetwork/lnd/lntest/node"
 	"github.com/lightningnetwork/lnd/lntest/wait"
 	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
 	"github.com/lightningnetwork/lnd/routing/route"
 	"github.com/stretchr/testify/require"
 )
 
+// testValidateMPPParams verifies that a payment using multipath parameters
+// fails with the expected error when the payment amount exceeds the allowed
+// maximum, and that when the multipath parameters permit the full payment
+// amount, the payment is successfully settled.
+func testValidateMPPParams(ht *lntest.HarnessTest) {
+	mts := newMppTestScenario(ht)
+
+	// Set up a network with three different paths Alice <-> Bob.
+	//
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob
+	//      \              /
+	//       \__ Dave ____/
+	//
+	paymentAmt := mts.setupSendPaymentCase()
+
+	// First, test that Alice's payment to Bob fails when the multipath
+	// parameters do not allow sending the full payment amount or requested
+	// max_parts exceeds the allowed upper limit.
+	testCases := []struct {
+		name            string
+		maxPartsInput   uint32
+		shardAmountMsat uint64
+		expectError     string
+	}{
+		{
+			name: "max_parts exceeds the allowed upper " +
+				"limit",
+			maxPartsInput:   1001,
+			shardAmountMsat: uint64(300_000),
+			expectError: "requested max_parts (1001) exceeds " +
+				"the allowed upper limit",
+		},
+		{
+			name: "required minimum max_parts exceeds the allowed" +
+				" upper limit",
+			maxPartsInput:   10,
+			shardAmountMsat: uint64(200_000),
+			expectError: "required minimum max_parts " +
+				"exceeds allowed limit",
+		},
+		{
+			name: "payment amount exceeds maximum " +
+				"allowed amount",
+			maxPartsInput:   10,
+			shardAmountMsat: uint64(300_000),
+			expectError: fmt.Sprintf("payment amount %v exceeds "+
+				"maximum allowed amount",
+				lnwire.MilliSatoshi(paymentAmt*1000)),
+		},
+	}
+
+	for _, tc := range testCases {
+		test := tc
+
+		success := ht.Run(test.name, func(t *testing.T) {
+			// Bob creates an invoice for the payment.
+			payReqs, _, _ := ht.CreatePayReqs(mts.bob, paymentAmt,
```

Comment:
I think we can instead create a single invoice for the whole test case as we know only the last send payment will succeed.